scalableminds/webknossos
# Rename resolution to mag in local variables

### URL of deployed dev instance (used for testing):
- https://renameresolutiontomaglocalvars.webknossos.xyz/

### Steps to test:
- test magnification slider when creating annotations or volume layers
- create skeleton nodes and save them
- export skeleton nodes and make sure the "inMag" field is set in the NML
- create a task type and restrict the mags
- within an annotation, zoom in and out
- do anything else that matters for mags.

### TODOs:
- [x] in-depth testing
     - [x] mag slider
    

- [x] write tests (see [slack conversation](https://scm.slack.com/archives/C5AKLAV0B/p1731399011397159) )
    - [x] one test to make sure the frontend CreateNodeAction matches the backend one (done by changing the mag in `generate_dummy_trees`)
    - [x] test that ServerNode and (backend) Node have the same fields (done by addind a typecheck file)


### some notes:
- I realized that changes in the frontend were possible, which were not only local variables. So I renamed the field "resolution" to "mag" in the following types:
  - [x] ServerNode 
  - [x] CreateNodeAction
  - [x] DataLayer 
- I did not edit the shader code
- I did not rename deprecated methods (e.g. data_cube.ts `_labelVoxelInResolution_DEPRECATED`)
- as these were not changed in the backend (see https://github.com/scalableminds/webknossos/pull/8111#issuecomment-2399313689), I did not change APIDataLayerBase.resolutions nor WKWDataLayer.wkwresolution


### Issues:
- contributes to #7378 

------
(Please delete unneeded items, merge only when none are left open)
